kratos/containers/edge_based_data_structure.h

```diff
@@ -677,7 +677,6 @@ class EdgeBasedDataStructure
 
         // Get position in the column indices vector as this is the same one to be used in the values vector
         const IndexType ij_col_index = GetColumVectorIndex(aux_i_id, aux_j_id);
-        KRATOS_ERROR_IF(ij_col_index < 0) << "Column index cannot be found for ij-edge " << GlobalIdI << "-" << GlobalIdJ << "." << std::endl;
```

Comment:
@rubenzorrilla this can never happen for an unsigned type

I think using `std::optional` would be nice here

Reply:
Sure. Most probably this is a leftover from a very first implementation or debugging. Let me merge this as it is to solve the Intel compiler issue and explore the `std::optional` in a separate branch.